core/src/macros.rs

```diff
@@ -405,4 +412,12 @@ mod tests {
         },));
         assert_eq!(h2, hlist![true, false, false]);
     }
+
+    #[test]
+    #[deny(clippy::unneeded_field_pattern)]
+    fn unneeded_field_pattern() {
+        let hlist_pat![_, _] = hlist![1, 2];
+        let hlist_pat![foo, _, baz] = hlist!["foo", "bar", "baz"];
+        let _ = (foo, baz);
```

Comment:
Is there any chance you could add an assertion to the values of the bindings? It would give us more confidence about the macro changes.